### ISSUE TITLE
Parse error on android-app://... URI

### ISSUE THREAD

**Issue by nielsbasjes** [nielsbasjes/logparser#28]
As described here: 
- https://firebase.google.com/docs/app-indexing/android/measure
- http://incredibly.co.uk/com-google-android-googlequicksearchbox-referral-google-analytics/

We find referrer values that look like these:

```
android-app://com.google.android.googlequicksearchbox
android-app://com.google.android.googlequicksearchbox/https/www.google.com
```

Which result in 

```
nl.basjes.parse.core.exceptions.DissectionFailure: Unable to parse the URI: >>>android-app://com.google.android.googlequicksearchbox<<< (unknown protocol: android-app)
nl.basjes.parse.core.exceptions.DissectionFailure: Unable to parse the URI: >>>android-app://com.google.android.googlequicksearchbox/https/www.google.com<<< (unknown protocol: android-app)
```

Place of the error is that the 
    at nl.basjes.parse.httpdlog.dissectors.HttpUriDissector.dissect(HttpUriDissector.java:138)

url = new URL(fieldValue);
which fails with a MalformedURLException with the message "unknown protocol: android-app"

**Comment by nielsbasjes** [comment]
I see two possible solution directions:

1. Register a custom protocol: http://stackoverflow.com/questions/26363573/registering-and-using-a-custom-java-net-url-protocol
2. Use a totally different way of parsing the URL

**Comment by nielsbasjes** [comment]
Switched from URL to URI solved the problem.
Fixed in version 2.5
